gtk4/src/subclass/buildable.rs

```diff
@@ -0,0 +1,384 @@
+        builder: &Builder,
+        child: &Object,
+        type_: Option<&str>,
+    );
+    fn parent_set_buildable_property(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        name: &str,
+        value: &Value,
+    );
+    fn parent_parser_finished(&self, buildable: &Self::Type, builder: &Builder);
+    fn parent_internal_child(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        name: &str,
+    ) -> Object;
+    fn parent_construct_child(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        name: &str,
+    ) -> Object;
+}
+
+impl<T: BuildableImpl> BuildableImplExt for T {
+    fn parent_set_id(&self, buildable: &Self::Type, id: &str) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
+                .set_id
+                .expect("no parent \"set_id\" implementation");
+
+            func(
+                buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+                id.to_glib_none().0,
+            )
+        }
+    }
+
+    fn parent_id(&self, buildable: &Self::Type) -> GString {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
+                .get_id
+                .expect("no parent \"get_id\" implementation");
+
+            from_glib_none(func(
+                buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+            ))
+        }
+    }
+
+    fn parent_add_child(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        child: &Object,
+        type_: Option<&str>,
+    ) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
+                .add_child
+                .expect("no parent \"add_child\" implementation");
+
+            func(
+                buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+                builder.to_glib_none().0,
+                child.to_glib_none().0,
+                type_.to_glib_none().0,
+            )
+        }
+    }
+
+    fn parent_set_buildable_property(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        name: &str,
+        value: &Value,
+    ) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
```

Comment:
i think we should just silently do nothing when there isn't a parent implementation.  I had to override it just because `gtk::Widget`  doesn't implement it. Probably the same for other `parent_*()` functions.

Reply:
Reading a little bit more of the docs. Set `buildable_property()` falls back to `g_object_set_property()` when there is no custom implementation. 

> Sets a property of a buildable object. It is normally not necessary to implement this, g_object_set_property() is used by default. GtkWindow implements this to delay showing itself (i.e. setting the “visible” property) until the whole interface is created.

Reply:
Should be better now :)

Reply:
I should have looked at gtkbuilder.c first ^^